packages/cdktf/lib/terraform-functions.ts

```diff
@@ -0,0 +1,1221 @@
+  }
+  return value;
+}
+
+function listOf(type: TFValueValidator): TFValueValidator {
+  return (value: any) => {
+    if (Tokenization.isResolvable(value)) {
+      return value;
+    }
+
+    if (!Array.isArray(value)) {
+      //   throw new Error(`${value} is not a valid list`);
+      return value;
+    }
+
+    value.forEach((item, i) => {
+      if (Tokenization.isResolvable(item)) {
+        return;
+      }
+
+      try {
+        type(item);
+      } catch (error) {
+        throw new Error(
+          `Element in list ${value} at position ${i} is not of the right type: ${error}`
+        );
+      }
+    });
+
+    return value;
+  };
+}
+
+function asString(value: IResolvable) {
+  return TokenMap.instance().registerString(value);
+}
+
+function asNumber(value: IResolvable) {
+  return TokenMap.instance().registerNumber(value);
+}
+
+function asList(value: IResolvable) {
+  return TokenMap.instance().registerList(value);
+}
+
+function terraformFunction(
+  name: string,
+  argValidators: TFValueValidator | TFValueValidator[]
+): ExecutableTfFunction {
+  return function (...args: any[]) {
+    if (Array.isArray(argValidators)) {
+      if (args.length !== argValidators.length) {
+        throw new Error(
+          `${name} takes ${argValidators.length} arguments, but ${args.length} were provided`
+        );
+      }
+      args.forEach((arg, i) => argValidators[i](arg));
+      return call(name, args);
+    } else {
+      return call(name, argValidators(args));
+    }
+  };
+}
+
+export class Fn {
+  /**
+   * {@link https://www.terraform.io/docs/language/functions/anytrue.html anytrue} returns true if all elements in a given collection are true or "true"
+   * @param {Array} values
+   */
+  public static alltrue(values: any[]) {
+    return terraformFunction("alltrue", [listOf(anyValue)])(values);
+  }
+
+  /**
+   * {@link https://www.terraform.io/docs/language/functions/anytrue.html anytrue} returns true if any element in a given collection is true or "true"
+   * @param {Array} value
+   */
+  public static anytrue(value: any[]) {
+    return terraformFunction("anytrue", [listOf(anyValue)])(value);
+  }
+
+  /**
+   * {@link https://www.terraform.io/docs/language/functions/chunklist.html chunklist} splits a single list into fixed-size chunks, returning a list of lists.
+   * @param {Array} value
+   * @param {number} chunkSize
+   */
+  public static chunklist(value: any[], chunkSize: number | IResolvable) {
+    return asList(
+      terraformFunction("chunklist", [listOf(anyValue), numericValue])(
+        value,
+        chunkSize
+      )
+    );
+  }
+
+  /**
+   * {@link https://www.terraform.io/docs/language/functions/coalesce.html coalesce} takes any number of arguments and returns the first one that isn't null or an empty string.
+   * @param {Array} value - Arguments are passed in an array
+   */
+  public static coalesce(value: any[]) {
+    return terraformFunction("coalesce", listOf(anyValue))(...value);
```

Comment:
Too bad we can't use generics to get more informative return type here.

Reply:
Yeah, it's really a bummer. I typed them as any now with a string representation 👍